### PR TITLE
Update response to HELLO command

### PR DESCRIPTION
**Pull request by alisaifee**
# Description
Update response from HELLO command to match the api of [redis](https://redis.io/commands/hello/). 
The `HELLO` command (and it's response) is important for clients that wish to provide the RESP3 protocol and is also a convenient way to get some initial information from the server such as client id and server version as the first handshake with the server (without making requests such as `INFO` which require a lot more parsing). 

## Notes:
- For any protocol version other than 2 an error matching what redis returns for `!{2,3}` is returned
- The array response is limited to 12 items since adding the value for `modules` seems unnecessary for now.
- No support / validation for authentication for now.

### CODE REVIEW

File: src/server/server_family.cc [hunk 1]
```diff
@@ -867,7 +867,29 @@ void ServerFamily::Info(CmdArgList args, ConnectionContext* cntx) {
...
+      (*cntx)->SendError("NOPROTO unsupported protocol version");
+      return;
+    }
+  }
+
+  (*cntx)->StartArray(12);
+  (*cntx)->SendBulkString("server");
+  (*cntx)->SendBulkString("redis");
+  (*cntx)->SendBulkString("version");
+  (*cntx)->SendBulkString(StrCat("df-", kGitTag));
```
Comment (romange) [inline src/server/server_family.cc:883]: `StrCat("df-", kGitTag)` is duplicated.
Can you please introduce `string GetVersion()` in `common.h` that does it?

**Review by romange — APPROVED** [review]
Thanks for fixing this! Added a minor comment.
In addition, I sent a PR #103  . hopefully it will be merged in few hours.
Please add this command to API 6 section in the doc

**Comment by romange** [comment]
One final thing - please add yourself to the contributors file if you wish.

File: src/server/server_family.cc (hunk 1 above)
Comment (romange) [reply src/server/server_family.cc:883]: Also, can you please add a simple test to `dragonfly_test.cc` that covers this functionality. It should be 3-4 lines, no more.

Comment (alisaifee) [reply src/server/server_family.cc:883]: Ah thanks for the pointers! Will do both shortly and ping you once it's ready for re-review.

Comment (alisaifee) [reply src/server/server_family.cc]: Addressed in a few follow up commits.

**Pull request merged**